File: 2020-10-30-client-libraries/python/idempotency/create_customer.py
# For more information about this demo, please see:
#   https://www.youtube.com/watch?v=94RgPhqBLfU
import stripe

stripe.api_key = "sk_test_xxx"

stripe.Customer.create(
  email="foo@bar.com",
  idempotency_key='a-long-random-string-20210115164236'
)

File: 2020-10-30-client-libraries/ruby/idempotency/create_customer.rb
# For more information about this demo, please see:
#   https://www.youtube.com/watch?v=dkpknsaWuQQ
require 'stripe'

Stripe.api_key = 'sk_test_xxx'

Stripe::Customer.create({
  email: 'foo@bar.com'
}, {idempotency_key: 'a-long-random-string-20210115164014'})
